### DISCUSSION TITLE
More macOS-like Taskbar UI (mockup included)

### DISCUSSION THREAD

**Discussion by jameszhan9592** [lawand-dot-io/taskbar#42] [Feature requests]
I believe Taskbar for Mac can benefit from a UI design that matches that of macOS's. While Mac's Dock lacks functionality compared to Windows' taskbar, I believe the former looks miles better than the latter.

I have attached a mockup showcasing a more macOS-like UI design for Taskbar for Mac just as an idea. I believe something like this will make Taskbar for Mac feel much more integrated to macOS.
![Frame 2](https://github.com/lawand-dot-io/taskbar/assets/76667111/32ff00ea-e200-41db-bb12-f5ec6b876d0a)

**Comment by lawand-dot-io** [comment]
Hello James. Thank you very much for taking the time to create the mock up. In general, the idea of making Taskbar as integrated as possible with macOS (visually and otherwise) is what I have in mind. Originally, I made the transparency low to make sure the window label text is readable. The dock is more transparent because it only has icons, but if you hover to see the app's name you see the same level of transparency that Taskbar currently has. I will experiment with this and see, perhaps I can add white outlines for the black labels so that they are still readable. As for making the width fit the tasks, that also makes sense but perhaps I should implement window addition/removal animation first, I'll also think about this. In any case, visual improvements should be implemented at some point. By the way I saw the email you sent me which includes this idea and other ideas. I have it in my to-do to go through that email. Cheers

**Reply by jameszhan9592** [reply]
Hi Lawand, thanks or the reply and for taking a look at my suggestion. Your thinking makes sense, and I'm glad to hear you will continue to think about this! I have been using Taskbar for Mac on a daily basis since the last update and it has been AWESOME. Can't wait for future updates :)

**Comment by livinlicious** [comment]
Your mockup looks cool but I would prefer if this is only a style option in the settings.
Currently the taskbar is in the bottom flush with the screen border. The mockup uses some spaces here that would be wasted on small screens like the 13inch internal. 
The simple styling as it is now is also nice especially since it takes the whole width of the screen from edge to edge and allows the windows to "sit" above the taskbar.
I can see this mockup design as an alternative design setting though.

**Reply by jameszhan9592** [reply]
Hi, thanks for taking a look at my mockup. 

> Currently the taskbar is in the bottom flush with the screen border. The mockup uses some spaces here that would be wasted on small screens like the 13inch internal.

I understand your point! My mockup is basically just the current macOS Dock UI altered with Taskbar for Mac features. The assets are all official ones from Apple. The gap between the Dock and the bottom of the screen in the mockup is the same size as it is in the default macOS Dock—it's part of the macOS design. 

I'm not a fan of the taskbar being flushed with the screen boarder as it feels way too Windows-like and I just don't like Windows' UI design, haha. It feels rather tight and cluttered to me.

Of course, ideally, Lawand could allow users to customize the gap from no gap to as much of a gap as the default macOS Dock has. I'm not sure how hard it would be to do that on the programming end, though!

> it takes the whole width of the screen from edge to edge

The mockup Dock is supposed to expand as there are more tabs in it, jus like how macOS's Dock expands when there are more app icons.

**Comment by livinlicious** [comment]
I understand your mockup, but again the 2 settings:
a) expanding taskbar
Is for people like me not so great, because that means every task is constant in motion. In the current setup and in Windows Styling, if you have 3 tasks they take the leftmost 3 "spacings" of the window. So your muscle/brain memory knows where the 3 open tasks are. Constantly size changing taskbar is a bit weird because you open and close different tabs.
Im ok to make taskbar size on a slider, so it can be a "floating" design. Thats cool. But dynamically changing size, im sure is not what people prefer. (but im not saying its bad, its just disorienting because positioning changes constantly)
pic to explain: 
<img width="1142" alt="2024-04-11+21-06-57" src="https://github.com/user-attachments/assets/7a8aa63a-9cad-44ab-94ee-884db1e5616f" />


b) spacing above and below
thats actually more of an issue. Since on 13inch this spacing is quite valuable.
Again, its fine if your design is live "floating style", and the current design is "block style", just in the styling preferences.
Anyone likes different things. Also I think yours would look nice when connected to big screen like 27inch display external. There you have the spacing. But removing the current style would be not the best.
This flush design is valuable I think because its so space efficient: 

<img width="933" alt="2024-04-11+21-05-07" src="https://github.com/user-attachments/assets/1962ada1-73d8-4f46-a0e5-ded4897e2cb0" />

Just my 2 cents. :)

**Reply by meirion** [reply]
I completely agree with you. The rigidity of layout in the Windows style task bar vs the constant flux and fluid size of the Mac OS dock is what makes the traditional Windows bar so much more straightforward to use, in my opinion.

With the Mac dock (or a centred design) it's changing width, locations are moving, icon sizes are increasing/decreasing.

**Reply by JJGITTY2018** [reply]
I second this, it will be nice to have the corners rounded, to fit the pill-like on Mac OS.

**Reply by jonathandpoynter** [reply]
I agree. There's a reason that sizing and position consistency of controls are a core part of usability heuristics: consistency & standards, recognition over recall etc.

**Comment by lawand-dot-io** [comment]
@jameszhan9592 @livinlicious thanks for the input :) I will think about this feature later on when I implement it, but I can most likely give users the option to have either of those two UIs and also most likely also allow configuring all spacings and gaps etc :)

**Reply by livinlicious** [reply]
options options options are king in UI, which apple seems to hate sadly.

**Reply by andreyrd** [reply]
I know everyone can have wildly differing opinions on styles, so I love that you plan on having configuration options. I for one would like to see more space under, around the taskbar, as well as a lot more space around the actual taskbar buttons.

**Reply by lawand-dot-io** [reply]
@andreyrd Yeah, for sure I would give the users the option since everybody has a different preference :)

**Comment by alexyshen** [comment]
I love it.  It's actually iOS 19 (VisionOS) like.  I would ditch that extra space below the bar though.  don't need it on the mac.  Thank you!

**Comment by lawand-dot-io** [comment]
Update: as a step towards this feature, the start menu beta includes a transparency option: https://x.com/lawand_dot_io/status/1993360744251789538

This will eventually be released as an official release

**Reply by lawand-dot-io** [reply]
The transparency preference is now included in v1.6.1 which is now released

**Reply by lawand-dot-io** [reply]
Note: This is a change in the default appearance of Taskbar but the old appearance can be restored by setting transparency to the lowest value in settings.

**Comment by RedSkyRoad** [comment]
May we please have round corners? The sharp corners clash with the OS look

**Reply by lawand-dot-io** [reply]
I originally made taskbar have sharp corners because it sits at the bottom of the screen and it matches the menu bar which is at the top of the screen and which also has sharp corners, as for the thumbnail view and the start menu I made both of them have rounded corners since they are inside the main part of the screen. I would consider adding an option in the future to make the taskbar itself rounded

**Reply by RedSkyRoad** [reply]
Tue combination of the round-cornered windows next to the sharp taskbar just looks weird. Is there an app that can make the window corners square?

**Reply by meirion** [reply]
>  Is there an app that can make the window corners square? 


Yes, it's called Windows. 🤣

**Reply by RedSkyRoad** [reply]
> > Is there an app that can make the window corners square?
> 
> Yes, it's called Windows. 🤣

What a sad individual, lol

**Reply by H852ing** [reply]
The best option is probably to get rid of those stupid rounded corners right away.
The fact that you can't change this on a Mac without a hack is just another example of Apple's arbitrary design choices.

The "**apple-sharpener**" project on Git looks quite promising.
